booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,659 @@
+    }
+    return !norNode.asBoolean(false);
+  };
+
+  private static final Consumer<MultiAttributeValidator> ALL_REQ_EQUIP = (mav) -> mav.submitAllMatching("requestedEquipments.*");
+  private static final JsonContentCheck NOR_PLUS_ISO_CODE_IMPLIES_ACTIVE_REEFER = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All requested Equipments where 'isNonOperatingReefer' is 'false' must have 'activeReeferSettings'",
+    ALL_REQ_EQUIP,
+    JsonAttribute.ifMatchedThen(
+      IS_ACTIVE_REEFER_SETTINGS_REQUIRED,
+      JsonAttribute.path("activeReeferSettings", JsonAttribute.matchedMustBePresent())
+    )
+  );
+
+  private static final JsonContentCheck IS_SHIPPER_OWNED_CONTAINER = JsonAttribute.customValidator(
+    "valid Requested Equipments",
+    (body) -> {
+      var issues = new LinkedHashSet<String>();
+      var requestedEquipments = body.path("requestedEquipments");
+      for (var rq: requestedEquipments) {
+        var isShipperOwned = rq.get("isShipperOwned").asBoolean(false);
+        var commodities = requestedEquipments.get("commodities");
+        if (isShipperOwned) {
+          for (var commodity : commodities) {
+            var cargoGrossWeight = commodity.get("cargoGrossWeight");
+            var cargoGrossWeightUnit = commodity.get("cargoGrossWeightUnit");
+            var cargoGrossVolume = commodity.get("cargoGrossVolume");
+              var cargoGrossVolumeUnit = commodity.get("cargoGrossVolumeUnit");
+            if (cargoGrossWeight == null || cargoGrossWeight.isEmpty()) {
+              issues.add("the container weight be provided");
+            }
+            if(cargoGrossWeightUnit == null || cargoGrossWeightUnit.isEmpty()) {
+              issues.add("the container weight unit must be provided");
+            }
+            if(cargoGrossVolume != null && (cargoGrossVolumeUnit  == null || cargoGrossVolumeUnit.isEmpty()) ) {
+              issues.add("the cargo gross volume unit must be provided");
+            }
+          }
+        }
+      }
+      return issues;
+    }
+  );
+
+
+  private static final JsonContentCheck COND_CARRIER_VOYAGE_NUMBER = JsonAttribute.customValidator(
+    "Conditional Carrier Export Voyage number",
+    (body) -> {
+      var expectedDepartureDate = body.path("expectedDepartureDate");
+      var expectedArrivalStartDate = body.path("expectedArrivalAtPlaceOfDeliveryStartDate");
+      var expectedArrivalEndDate = body.path("expectedArrivalAtPlaceOfDeliveryStartDate");
+      var carrierExportVoyageNumber = body.path("carrierExportVoyageNumber");
+      var issues = new LinkedHashSet<String>();
+
+      if (carrierExportVoyageNumber == null || (expectedDepartureDate == null || ( expectedArrivalEndDate == null && expectedArrivalStartDate == null)))  {
+        issues.add("The carrierExportVoyageNumber must be present as either expectedDepartureDate or expectedArrivalDates are not present");
+      }
+
+      return issues;
+    }
+  );
```

Comment:
Please provide an `atLeastOneOf` method in the `JsonAttribute` (you can use `atMostOneOf` from the latest dev branch as a starting point) and rewrite this check to use `atLeastOneOf`.

Reply:
Nvm, this is not a true "at least one of"